ring/http.go

```diff
@@ -0,0 +1,101 @@
+package ring
+
+import (
+	"fmt"
+	"html/template"
+	"net/http"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+)
+
+const tpl = `
+<!DOCTYPE html>
+<html>
+	<head>
+		<meta charset="UTF-8">
+		<title>Cortex Ring Status</title>
+	</head>
+	<body>
+		<h1>Cortex Ring Status</h1>
+		<p>{{ .Message }}</p>
+		<form action="" method="POST">
+			<table width="100%" border="1">
+				<thead>
+					<tr>
+						<th>Ingester</th>
+						<th>State</th>
+						<th>Address</th>
+						<th>Last Heartbeat</th>
+						<th>Actions</th>
+					</tr>
+				</thead>
+				<tbody>
+					{{ range $key, $value := .Ring.Ingesters }}
+					<tr>
+						<td>{{ $key }}</td>
+						<td>{{ $value.State }}</td>
+						<td>{{ $value.Hostname }}</td>
+						<td>{{ $value.Timestamp | time }}</td>
```

Comment:
IIRC, this time is from the ingester's clock, but the state is inferred from the distributor's clock. 

Either way, this page should either show the distributor's opinion of the current time outside the table (so we have a basis for comparison), or the delta between the current time and the last heartbeat in the table.